profiler/src/main/java/com/splunk/opentelemetry/profiler/snapshot/PeriodicStackTraceSampler.java

```diff
@@ -90,8 +104,7 @@ public void close() {
   }
 
   private static class ThreadSampler extends Thread {
-    private final Map<SpanContext, SamplingContext> traceSamplingContexts =
-        new ConcurrentHashMap<>();
+    private final Map<Thread, SamplingContext> threadSamplingContexts = new ConcurrentHashMap<>();
```

Comment:
Wondering whether we should use a weak map to allow collecting threads even when they weren't removed from the map. I guess that would mean that `stop` wasn't called and there would be other issues as well, so probably not worth the effort.